ehr/test/src/org/labkey/test/tests/ehr/AbstractGenericEHRTest.java

```diff
@@ -300,6 +306,160 @@ private void testUserAgainstAllStates(@LoggedParam EHRUser user)
+                "ehr-cacheLivingAnimals.view",
+                "core-modulePropertyAdmin.view",
+                "dataintegration-begin.view",
+                "ldk-updateQuery",
+                "junit-begin.view"
+        );
+    }
+
+    private String validLink(WebElement anchor)
+    {
+        boolean clickable = false;
+        String validUrl = null;
+
+        String href = anchor.getDomAttribute("href");
+        if (href != null && !href.startsWith("#"))
+        {
+            String decodedHref = EscapeUtil.decodeUriPath(href);
+            if (skipLinksForValidation().stream().anyMatch(s -> decodedHref.toLowerCase().contains(s.toLowerCase())))
+            {
+                log(href + " is specified as an exception to link validation. Skipping validation.");
+                return validUrl;
+            }
+
+            // Ensure link is not external
+            try
+            {
+                URL url = new URL(href);
+                if (!url.getHost().equalsIgnoreCase(getURL().getHost()))
+                {
+                    log(href + " is an external link. Skipping validation.");
+                    return validUrl;
+                }
+            }
+            catch (MalformedURLException e)
+            {
+                // not a full URL so not external. Carry on.
+            }
+
+            // scope this to admin, ehr folder and subfolders
+            if (!decodedHref.contains(getContainerPath()) && !decodedHref.startsWith("/admin"))
+            {
+                log(href + " is in a different folder than the EHR folder, " + getContainerPath() + ". Skipping validation.");
+                return validUrl;
+            }
+        }
+
+        if (anchor.isDisplayed() && anchor.isEnabled())
+        {
+            clickable = true;
+
+            try
+            {
+                openLinkInNewWindowOrThrow(anchor);
+            }
+            catch (WebDriverException | IllegalStateException e)
+            {
+                clickable = false;
+            }
+
+            if (clickable)
+            {
+                // Give page time to load
+                boolean loaded = waitFor(() -> (getDriver().getCurrentUrl() != null && !getDriver().getCurrentUrl().equalsIgnoreCase("about:blank")), WAIT_FOR_PAGE);
+                assertTrue("Link " + href + " did not load in " + WAIT_FOR_PAGE + "ms.", loaded);
+
+                // Assert page is not empty and does not have errors
+                URL url = getURL();
+                assertFalse("URL " + url + " is empty.", isPageEmpty());
+                assertNoLabKeyErrors();
+
+                // assertNoLabKeyErrors does not catch all types of errors
+                assertElementNotPresent("LabKey error found for URL " + url, Locators.labkeyErrorHeading);
+
+                // record link as valid and cleanup
+                validUrl = url.toString();
+                switchToWindow(0);
+                quietlyCloseExtraWindows();
```

Comment:
No need to use a special method to close all of the windows. There's only one to close.
```suggestion
                // record link as valid and cleanup
                getDriver().close();
                switchToWindow(0);
                return url.toString();
```

Reply:
Nice I can get rid of quietlyCloseExtraWindows